### DISCUSSION TITLE
Filament v2 - Form Builder: Is hiding a Select (relationship) safe enough to avoid tampering?

### DISCUSSION THREAD

**Discussion by emcarson** [filamentphp/filament#7073] [Help] [answered]
Hello everyone, sorry if it's a stupid question, but I can't seem to figure it out.

I'm looking to saving a relationship on create only, I want to ensure 100% that the user can't modify that field on update.

I'm currently hiding the field conditionally, but I'm not really sure if that is enough to avoid tampering.

```php
Forms\Components\Select::make('role_id')
    ->relationship('role', 'name')
    ->hidden(fn (string $context): bool => $context === 'edit'),
```

I seem to figured it out with TextInputs, but I don't know the equivalent for relationships:

```php
Forms\Components\TextInput::make('commission')
    ->disabled(fn (string $context): bool => $context === 'edit')
    ->dehydrated(fn (string $context): bool => $context === 'create'),
```

Thank you.

PS: I tried using a combination of disabled / dehydrated / saveRelationshipUsing, but it's quite cumbersome (lots of repetition) and only works if I define both "dehydrated" AND "saveRelationshipUsing".

Also, I have no idea how to set the correct "saveRelationshipUsing" for create.

```php
Forms\Components\Select::make('role_id')
    ->relationship('role', 'name')
    ->disabled(fn (string $context): bool => $context === 'edit')
    ->dehydrated(fn (string $context): bool => $context === 'create')
    ->saveRelationshipsUsing(null), // What should I set here for creation?
```

**Comment by mstfkhazaal** [comment, marked as answer]
You can use mutateFormDataBefore(Create/Edit)
https://filamentphp.com/docs/2.x/admin/resources/creating-records#customizing-data-before-saving

**Comment by emcarson** [comment]
Thank you @mstfkhazaal, that worked great!

For anyone interested, I added this to my Edit resource class:

```php
protected function mutateFormDataBeforeSave(array $data): array
{
    unset($data['role_id']);

    return $data;
}
```

The only one I couldn't figure out yet is how to disable modifying a BelongsToMany relationship on edit (Multiple Select).

**Reply by mstfkhazaal** [reply]
to disable modifying a BelongsToMany relationship on edit
you can remove relationship 
an add ->options(fn()=>Role::all()->pulck('name','id');

you can read this documentation 
https://filamentphp.com/docs/2.x/forms/fields#select

**Reply by emcarson** [reply]
That's amazing, thanks a lot!

I started my project using Laravel Nova and found it really buggy and cumbersome, I couldn't stand working with it anymore.
Last week I installed FilamentAdmin for the first time and I already did in a week what took me 2 months with Nova.
It's really great what you have created.
